### ISSUE TITLE
x/build: coordinator updates for coreos changes

### ISSUE THREAD

**Issue by bradfitz** [golang/go#12631]
I updated to a new version of coreos on the farmer. (It had been ages)

I now notice that curl returns 23 or something if the file already exists, so sometimes the systemd unit fails to update unless I `rm /opt/bin/coordinator` first. Fix the systemd unit's shell line.

Also: I see problems bringing up the watcher now sometimes:

```
============== Watcher https://go.googlesource.com/go, last fail:
2015/09/15 21:17:27 Running watcher role.
2015/09/15 21:17:27 Watcher exiting after failure: listen tcp 127.0.0.1:21536: bind: address already in use


============== Watcher https://go.googlesource.com/go, current container logs:
time="2015-09-15T21:17:41Z" level=fatal msg="Error response from daemon: no such id: f01634497cc75822001481f0cfd0e6d1bd9fae92625c9b367c3492fa51d4291f" 

```

And sure enough, it's running, but not from an earlier process. We should verify that only our generation's watcher is running, and kill it otherwise.

/cc @adg

**Comment by bradfitz** [comment]
To @adg in chat:

> 23     Write error. Curl couldn't write data to a local filesystem or similar
> but only started after coreos update
> oh!
> I know.
> it's "Text file busy", now that the executable binary is now also used by the watcher process in the > > docker container
> easy fix
> gotta use install(1)
> not curl -o

**Comment by bradfitz** [comment]
Fixed by 4483ca5 and d79d77f945f5d
